pkg/daemon/ceph/cephconfig/config.go

```diff
@@ -0,0 +1,275 @@
+/*
+Ceph daemon config package provides methods for creating and formatting Ceph configuration files.
+*/
+
+package cephconfig
```

Comment:
To be consistent with go naming conventions, I'd suggest we just call this package `config`. This would remove the redundancy from the package name `pkg/daemon/ceph/config`. If this leads to a conflict in common package names, the consumer of the package can declare the import with a non-conflicting name like so:
```
   cephconfig "github.com/rook/rook/pkg/daemon/ceph/config"
```

Reply:
I had similar reservations. I also had a thought that the conflicting (or ambiguous) `config` is often a struct like in `cmd/rook/ceph/ceph.go` or `pkg/daemon/ceph/mon/daemon.go`. I see good reason to rename `ceph/cephconfig` to `ceph/config`. I would, for clarity of reading the code, prefer to change all instances of the include to `cephconfig "github.com/rook/rook/pkg/daemon/ceph/config"` and not just conflicting ones. I think doing so makes it easier to understand when looking at the line `cephconfig.GetConfFilePath` that it is the path of the ceph conf file versus `conf.GetConfFilePath` which could be the path of any conf file unless you scroll to the top of the file to see the `include` directives.

Coming to the Rook code fresh, I feel like the different `config`/`Config` items make it hard to see what is going on easily without having to always look at the includes for full context. I also understand the desire to not be repetitive about naming. As I've been thinking about it more, I think a middle balance exists somewhere. An idea I had would be to include the `pkg/daemon/ceph/mon` via `mondaemon "github.com/rook/rook/pkg/daemon/mon"` which would then make the config access `mondaemon.Config` (versus `mon.Config`). I would find this easier to understand by reading an individual line, and I personally don't think it adds unnecessary verbosity. I've found it confusing a few times trying to remember which `mon` package the particular file I'm working with refers to, whether the `pkg/daemon` one or the `cmd` one.

How would you feel, @travisn, about my changing `daemon/mon` includes to `mondaemon "....../daemon/mon"` here?

Reply:
I understand the confusion about where the packages are coming from, since there are similar package names under `cmd`, `daemon`, and `operator`. I've had to look at the includes many times myself to disambiguate. Agreed that it will help to consume them with the more descriptive name such as `cephconfig` and `mondaemon`.